### PR TITLE
[WFLY-14375] Upgrade Apache DS to version AM26

### PR DESCRIPTION
**Pull request by rmartinc**
Issue: https://issues.redhat.com/browse/WFLY-14375

The upgrade needs the following:

* The dependency now is `apacheds-test-framework` and `apacheds-interceptor-kerberos` is also required for the tests that need the kerberos interceptor.
* Two new opens are added for jdk-17.
* Some changes in test classes to accommodate to the new API (eh-cache is not used anymore for example).
* I think that the modifications in `Policy.java` just fix a mistake, a eh-cache class was included for javadoc instead of the elytron one.

### CODE REVIEW

**Comment by rmartinc** [comment]
@fjuma This is the upgrade for apache-ds. Take a look when you have time.

**Comment by darranl** [comment]
I have just tried running this after merging to big-bang locally but I am seeing test failures with NoClassDefFound errors.  

I am just syncing main to big-bang - once my changes are merged would it be possible for you to rebase this on big-bang and submit a PR against that branch instead?

https://github.com/wildfly/wildfly/pull/15724

**Comment by rmartinc** [comment]
Yes, no problem @darranl, I'll check it when the other PR is merged.

File: testsuite/integration/basic/pom.xml [hunk 1]
```diff
@@ -41,7 +41,7 @@
         <bootable-jar-copy-module-files.phase>none</bootable-jar-copy-module-files.phase>
         <!-- Disable the default surefire test execution. -->
         <surefire.default-test.phase>none</surefire.default-test.phase>
-        <modular.jdk.testsuite.args>${modular.jdk.args} --add-opens=java.base/javax.crypto=ALL-UNNAMED --add-opens=java.base/sun.security.validator=ALL-UNNAMED</modular.jdk.testsuite.args>
+        <modular.jdk.testsuite.args>${modular.jdk.args} --add-opens=java.base/javax.crypto=ALL-UNNAMED --add-opens=java.base/sun.security.validator=ALL-UNNAMED --add-opens=java.base/sun.security.x509=ALL-UNNAMED --add-opens=java.base/sun.security.util=ALL-UNNAMED</modular.jdk.testsuite.args>
```
Comment (ropalka) [inline testsuite/integration/basic/pom.xml:44]: LGTM

File: testsuite/pom.xml [hunk 2]
```diff
@@ -160,7 +160,7 @@
         <surefire.memory.args>-Xmx512m -XX:MetaspaceSize=128m</surefire.memory.args>
         <surefire.jpda.args></surefire.jpda.args>
         <as.debug.port>8787</as.debug.port>
-        <modular.jdk.testsuite.args>${modular.jdk.args}</modular.jdk.testsuite.args>
+        <modular.jdk.testsuite.args>${modular.jdk.args} --add-opens=java.base/sun.security.x509=ALL-UNNAMED --add-opens=java.base/sun.security.util=ALL-UNNAMED</modular.jdk.testsuite.args>
```
[the remark this answers is not in the document]
Comment (ropalka) [reply testsuite/pom.xml:163]: If these 'add opens' are needed across different testsuite maven subprojects then testsuite/pom.xml is the right place to configure them. If only testsuite/integration/basic, then these 'add opens' clauses should be there only @rmartinc

[the remark this answers is not in the document]
Comment (rmartinc) [reply testsuite/pom.xml:163]: Finally only `integration/basic` and `integration/elytron` use the apache-ds . The references in `integration/manualmode` are leftovers that I'm going to delete if you don't say something different. So I'm going to just add the opens in the two folders.

**Comment by rmartinc** [comment]
@darranl Doing more tests now but I don't see any NoClassDefFound in the PR. Nevertheless I detected that there are leftovers that reference apache-ds classes since WFLY-15001 (https://github.com/wildfly/wildfly/pull/14584) in the `manualmode` folder (for example [these files](https://github.com/wildfly/wildfly/tree/27.0.0.Alpha1/testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/security)). In that PR the tests `OutboundLdapConnectionTestCase` and `OutboundLdapConnectionClientCertTestCase` were removed but related files remain with no use at all. I'm going to delete them in this PR if you don't say something different.

**Comment by darranl** [comment]
@rmartinc I can have another look, there were some other issues popped up yesterday I needed to resolve so maybe not an issue after all.

**Comment by darranl** [comment]
I need to check some more in case it is my environment but I am seeing the following error when I test after merging to big bang:

Caused by: javax.naming.CommunicationException: 127.0.0.1:10389 [Root exception is java.net.ConnectException: Connection refused (Connection refused)]
    at java.naming/com.sun.jndi.ldap.Connection.<init>(Connection.java:252)
    at java.naming/com.sun.jndi.ldap.LdapClient.<init>(LdapClient.java:137)
    at java.naming/com.sun.jndi.ldap.LdapClient.getInstance(LdapClient.java:1616)

**Comment by darranl** [comment]
I have just checked the same test when this PR is merged to main and don't get a failure, if you would not mind looking at these tests once merged to big-bang as otherwise this will prevent us syncing main to big-bang.

**Comment by rmartinc** [comment]
@darranl Probably the error is because big-bang branch has [this lines](https://github.com/wildfly/wildfly/blob/big-bang/testsuite/integration/basic/pom.xml#L246-L250) in `basic` with apache-ds again (this is not in main branch). The artifact should be changed to `apacheds-test-framework`. But don't worry. This can wait.

**Comment by darranl** [comment]
I am just adding a hold label for now so we can coordinate with big-bang - alternatively if you were able to submit directly to big bang we could merge there instead of to main.

**Comment by rmartinc** [comment]
@darranl I have submitted the same for big-bang in #15738. So you can close this and continue with the big-bang PR if you want. But not urgent, so I can wait too. Thanks!

**Comment by darranl** [comment]
Merged to big-bang under https://github.com/wildfly/wildfly/pull/15738 - big-bang will later be merged back to main so I will close this PR.

**Pull request closed without merging**